### PR TITLE
Add Mac compatability.

### PR DESCRIPTION
**Pull request by johnrtipton**
DRAFT:
Add Mac compatibility, I need to do some cleanup still, this is a rough implementation.

### CODE REVIEW

File: glados/tts.py [hunk 1]
```diff
@@ -216,9 +218,11 @@ class Phonemizer:
     espeakVOICE = "en-us"
 
     def __init__(self):
-        self.libc = ctypes.cdll.LoadLibrary("libc.so.6")
+        # self.libc = ctypes.cdll.LoadLibrary("libc.so.6")
+        self.libc = ctypes.cdll.LoadLibrary("libc.dylib")
         self.libc.open_memstream.restype = ctypes.POINTER(ctypes.c_char)
-        self.lib_espeak = self._load_library("libespeak-ng.so", "libespeak-ng.so.1")
+        #self.lib_espeak = self._load_library("libespeak-ng.so", "libespeak-ng.so.1")
+        self.lib_espeak = self._load_library("libespeak.dylib", "libespeak.1.dylib")
```
Comment (Traxmaxx) [inline glados/tts.py:225]: What worked for me on MacOS was
```
import platform
[...]
self.lib_espeak = ctypes.cdll.LoadLibrary("libespeak-ng.{}".format("so.1" if platform.uname()[0] != "Darwin" else "dylib"))
```

Still need to test on linux, will do later today

**Comment by Traxmaxx** [comment]
Installing `espeak-ng` through homebrew and doing the mentioned change above, espeak is not directly crashing and works for some time until I run into mentioned error. I'm still investigating what is happening there.
Any clues?

**Comment by fonix232** [comment]
I've been doing some refactor locally, which I think could go into this PR.

Instead of relying on manually compiled and installed libraries, my suggestion would be an init script that:

1. Enters a Python venv
2. Checks for libwhisper, if not found, pulls the git repo, builds it, and copies the appropriate files into a local path
3. Checks for libespeak-ng, if not found, pulls the git repo, builds it, and copies the appropriate files into a local path
4. Checks for llama.cpp, and you guessed it right, if not found, pulls the git repo, builds it, and copies the appropriate files into a local path
5. Installs all Python dependencies

Optionally it could also prompt for installing the `make`/`cmake`/etc. build tools as needed, though with all the different distros, this might be a task a smidge too big.

I'd also go for masking `ctypes.cdll.LoadLibrary` with a custom method that takes care of the `dylib` or `so` checking (note that e.g. libwhisper _still_ builds a `.so` file on macOS, so on `Darwin` platforms it's best to check first for `dylib` then for `so` and only if neither are found, throw the exception).

**Comment by fonix232** [comment]
@dnhkng please see my above comment to expand on the discussion under #11

**Comment by dnhkng** [comment]
We now use the espeak-ng binary, as it resolves the segfault issues, so I am closing this,

**Pull request closed without merging**